ament_tools/build_types/ament_cmake.py

```diff
@@ -153,11 +153,21 @@ def on_test(self, context):
         assert context.build_tests
         # Figure out if there is a setup file to source
         prefix = self._get_command_prefix('test', context)
-        if has_make_target(context.build_space, 'test') or context.dry_run:
-            yield BuildAction(prefix + [MAKE_EXECUTABLE, 'test'])
+        if not IS_WINDOWS:
+            if has_make_target(context.build_space, 'test') or context.dry_run:
+                yield BuildAction(prefix + [MAKE_EXECUTABLE, 'test'])
+            else:
+                self.warn("Could not run test for ament_cmake package because it "
+                          "has no 'test' target")
         else:
-            self.warn("Could not run test for ament_cmake package because it "
-                      "has no 'test' target")
+            if MSBUILD_EXECUTABLE is None:
+                raise VerbExecutionError("Could not find 'msbuild' executable")
+            run_tests_project_file = project_file_exists_at(context.build_space, 'RUN_TESTS')
+            if run_tests_project_file is not None or context.dry_run:
+                yield BuildAction(prefix + [MSBUILD_EXECUTABLE, run_tests_project_file])
+            else:
+                self.warn("Could not locate RUN_TESTS VS project file, "
+                          "did you build with the --build-tests option?")
```

Comment:
The `test` verb builds upon the the `test_pkg` verb (https://github.com/ament/ament_tools/blob/07821eb659c3ff9d7390ec9584730191c845ecb1/ament_tools/verbs/test/cli.py#L56). And the `test_pkg` verb always sets `build_tests` (https://github.com/ament/ament_tools/blob/07821eb659c3ff9d7390ec9584730191c845ecb1/ament_tools/verbs/test_pkg/cli.py#L28). Otherwise the `assert` would not be `True` all the time.

Arguably the `test*` verbs could remove the `--build-tests` option from argparse since it will always be set for them.